apis/Google.Cloud.Vision.V1/Google.Cloud.Vision.V1/AnnotationResult.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using Google.Protobuf.Collections;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Google.Cloud.Vision.V1
+{
+    // TODO: For a single image detecting a single annotation type, will there ever be an error *and* results?
+    // Should we actually just throw an exception?
+
+    /// <summary>
+    /// The annotations associated with a single annotation type (such as faces, or landmarks)
+    /// from one image, along with any error status returned as part of the response.
+    /// Any annotations returned are valid, even if an error message is present.
+    /// </summary>
+    /// <typeparam name="T">The type of annotation returned.</typeparam>
+    public sealed class AnnotationResult<T> : IReadOnlyList<T>
+    {
+        private readonly RepeatedField<T> annotations;
+
+        /// <summary>
+        /// The error status in the response, or <c>null</c> if the call was successful.
+        /// </summary>
+        public Rpc.Status Error { get; }
```

Comment:
This order doesn't seem quite right, should this property be after the constructor?